hoomd/md/EvaluatorPairTable.h

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2009-2021 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+
+// Maintainer: joaander
+
+#include "hoomd/ForceCompute.h"
+#include "NeighborList.h"
+#include "hoomd/Index1D.h"
+#include "hoomd/GlobalArray.h"
+#include "hoomd/ManagedArray.h"
+#include <memory>
+
+/*! \file TablePotential.h
+    \brief Declares the TablePotential class
+*/
+
+#ifdef __HIPCC__
+#error This header cannot be compiled by nvcc
+#endif
+
+// need to declare these class methods with __device__ qualifiers when building in nvcc
+#ifdef __HIPCC__
+#define DEVICE __device__
+#else
+#define DEVICE
+#endif
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#ifndef __TABLEPOTENTIAL_H__
+#define __TABLEPOTENTIAL_H__
+
+//! Computes the potential and force on each particle based on values given in a table
+/*! \b Overview
+    Pair potentials and forces are evaluated for all particle pairs in the system within the given cutoff distances.
+    Both the potentials and forces** are provided the tables V(r) and F(r) at discreet \a r values between \a rmin and
+    \a rmax. Evaluations are performed by simple linear interpolation, thus why F(r) must be explicitly specified to
+    avoid large errors resulting from the numerical derivative. Note that F(r) should store - dV/dr.
+
+    \b Table memory layout
+
+    V(r) and F(r) are specified for each unique particle type pair. Three parameters need to be stored for each 
+    potential: rmin, rmax, and dr, the minimum r, maximum r, and spacing between r values in the table respectively. 
```

Comment:
I think rmax is misleading in these comments. There is no rmax variable stored or computed in the new code. I would update the documentation to refer to r_cut instead of rmax.